src/rust/engine/src/downloads.rs

```diff
@@ -44,11 +46,21 @@ impl NetDownload {
   async fn start(
     core: &Arc<Core>,
     url: Url,
+    auth_headers: BTreeMap<String, String>,
     file_name: String,
   ) -> Result<NetDownload, StreamingError> {
+    let mut headers = HeaderMap::new();
+    for (k, v) in auth_headers.iter() {
```

Comment:
More idiomatic would be `for (k, v) in &auth_headers`. No need to call `.iter()` usually with a `for ... in` loop in Rust.